src/main/java/com/robothand/highqualitybot/permission/PermissionManager.java

```diff
@@ -0,0 +1,98 @@
+package com.robothand.highqualitybot.permission;
+
+import com.robothand.highqualitybot.Bot;
+import com.robothand.highqualitybot.command.Command;
+import com.robothand.highqualitybot.command.Commands;
+import net.dv8tion.jda.core.entities.Member;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+
+/**
+ * Handles PermissionGroups, and provides methods for checking a user's permissions on a certain action.
+ */
+public class PermissionManager {
+    private static final PermissionManager instance = new PermissionManager();
+    private ArrayList<PermissionGroup> groups;
+
+    private PermissionManager() {
+        groups = new ArrayList<>();
+    }
+
+    public static PermissionManager instance() {
+        return instance;
+    }
+
+    // parses a list of command names separated by commas into an ArrayList of Commands
+    public ArrayList<Command> parseCommandList(String list) {
+        ArrayList<Command> commands = new ArrayList<>();
+
+        for (String name : list.split(",")) {
+            if (name.trim().equals("*")) {
+                commands = Commands.getInstance().getCommands();
+                break;
+            }
+
+            Command command = Commands.getInstance().getCommand(name.trim());
+            if (command != null) {
+                commands.add(command);
+            }
+        }
+
+        return commands;
+    }
+
+    public boolean hasPermission(Member member, Command command) {
+        // check for owner
+        if (member.getUser().getId().equals(Bot.config.OWNERID)) {
+            return true;
+        }
+
+        boolean permission = false;
+
+        // start with default permissions
+        int def = 0;
+        if (Bot.config.ALLOWED.contains(command)) {
+            def = 1;
+        }
+
+        if (Bot.config.DISALLOWED.contains(command) && (def == 0 || !Bot.config.ALLOWEDHASPRECEDENCE)) {
+            def = -1;
+        }
+
+        switch (def) {
```

Comment:
Same as the switch in PermissionGroup.hasPermission(). It's a bit redundant here, I'm going to get rid of it